src/Shape/Box.js

```diff
@@ -0,0 +1,385 @@
+      .render();
+
+    // Draw median.
+    new Rect()
+      .data(filteredData)
+      .x(d => orient === "vertical" ? d.x : d.medianX)
+      .y(d => orient === "vertical" ? d.medianY : d.y)
+      .height(d => orient === "vertical" ? 1 : d.height)
+      .width(d => orient === "vertical" ? d.width : 1)
+      .select(elem("g.d3plus-Box-Median", {parent: this._select}).node())
+      .config(configPrep.bind(this)(this._medianConfig, "shape"))
+      .render();
+
+    // Draw 2 lines using Whisker class.
+    // Construct coordinates for whisker startpoints and push it to the whiskerData.
+    const whiskerData = [];
+    filteredData.forEach((d, i) => {
+      const x = d.x;
+      const y = d.y;
+      const topLength = d.top - d.third;
+      const bottomLength = d.first - d.bottom;
+
+      if (orient === "vertical") {
+        const topY = y - d.height / 2;
+        const bottomY = y + d.height / 2;
+        whiskerData.push(
+          {__d3plus__: true, data: d, i, x, y: topY, length: topLength, orient: "top"},
+          {__d3plus__: true, data: d, i, x, y: bottomY, length: bottomLength, orient: "bottom"}
+        );
+      }
+      else if (orient === "horizontal") {
+        const topX = x + d.width / 2;
+        const bottomX = x - d.width / 2;
+        whiskerData.push(
+          {__d3plus__: true, data: d, i, x: topX, y, length: topLength, orient: "right"},
+          {__d3plus__: true, data: d, i, x: bottomX, y, length: bottomLength, orient: "left"}
+        );
+      }
+
+    });
+
+    // Draw whiskers.
+    new Whisker()
+      .data(whiskerData)
+      .select(elem("g.d3plus-Box-Whisker", {
+        parent: this._select
+      }).node())
+      .endpointConfig({Rect: {
+        height: () => orient === "vertical" ? 5 : 20,
+        width: () => orient === "vertical" ? 20 : 5
+      }})
+      .config(configPrep.bind(this)(this._whiskerConfig, "shape"))
+      .render();
+
+    // Draw outliers.
+    nest()
+      .key(d => d.outlier)
+      .entries(outlierData)
+      .forEach(shapeData => {
+        const shapeName = shapeData.key;
+        new shapes[shapeName]()
+          .data(shapeData.values)
+          .select(elem(`g.d3plus-Box-Outlier-${shapeName}`, {parent: this._select}).node())
+          .config(configPrep.bind(this)(this._outlierConfig, "shape", shapeName))
+          .render();
+      });
+
+    return this;
+  }
+
+  /**
+      @memberof Box
+      @desc If *data* is specified, sets the data array to the specified array and returns the current class instance. If *data* is not specified, returns the current data array.
+      @param {Array} [*data* = []]
+      @chainable
+  */
+  data(_) {
+    return arguments.length ? (this._data = _, this) : this._data;
+  }
+
+  /**
+      @memberof Box
+      @desc If *value* is specified, sets the id accessor to the specified function and returns the current class instance.
+      @param {Function} [*value*]
+      @chainable
+  */
+  id(_) {
```

Comment:
Now that we are using `x` or `y` for the data nesting, this `id` method and `this._id` are not needed.